src/decoder/mod.rs

```diff
@@ -895,6 +898,18 @@ impl<R: Read + Seek> Decoder<R> {
         self.get_tag(tag)?.into_string()
     }
 
+    pub fn tag_iter(&mut self) -> impl Iterator<Item = TiffResult<(Tag, ifd::Value)>> + '_ {
+        match self.image().ifd.as_ref() {
+            None => Either::Left(std::iter::empty()),
```

Comment:
I have no idea, honestly. I think from a developer perspective, it is more convenient to have an empty iterator instead of having to deal with an `Option`.

Reply:
Ok, I went through the existing code. It should be impossible for `ifd` to be None here, so please just `.unwrap()` it